batch/create/create_with_runnables_labels.py

```diff
@@ -0,0 +1,121 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import google.auth
+
+# [START batch_labels_runnable]
+from google.cloud import batch_v1
+
+
+def create_job_with_custom_runnables_labels(
+    project_id: str,
+    region: str,
+    job_name: str,
+    label_key1: str,
+    label_value1: str,
+    label_key2: str,
+    label_value2: str,
+) -> batch_v1.Job:
+    """
+    This method creates a Batch job with custom labels for runnables.
+    Args:
+        project_id (str): project ID or project number of the Cloud project you want to use.
+        region (str): name of the region you want to use to run the job. Regions that are
+            available for Batch are listed on: https://cloud.google.com/batch/docs/locations
+        job_name (str): the name of the job that will be created.
+        label_key1 (str): First custom label key.
+        label_value1 (str): First custom label value.
+        label_key2 (str): Second custom label key.
+        label_value2 (str): Second custom label value.
+    Returns:
+        batch_v1.Job: The created Batch job object containing configuration details.
+    """
+    client = batch_v1.BatchServiceClient()
+
+    runn1 = batch_v1.Runnable()
+    runn1.display_name = "Script 1"
+    runn1.script = batch_v1.Runnable.Script()
+    runn1.script.text = "echo Hello world from Script 1 for task ${BATCH_TASK_INDEX}"
+    # Add custom labels to the first runnable
+    runn1.labels = {label_key1: label_value1}
+
+    runn2 = batch_v1.Runnable()
+    runn2.script = batch_v1.Runnable.Script()
+    runn2.script.text = "echo Hello world!"
+    # Add custom labels to the second runnable
+    runn2.labels = {label_key2: label_value2}
```

Comment:
In this case, I used two scripts in `Runnables`. The reason for this is that using a `Container` together with a `Script` results in the Job ending with a status of `FAILED`.
I tested the following scenarios:
One `Runnable` with a container - works.
Two `Runnables` with a container and a Script - Batch Job ends with a status of `FAILED`.

Additionally, I am attaching the `config.json` file presented in the [documentation](https://cloud.google.com/batch/docs/organize-resources-using-labels). Creating a job using this configuration file also results in a `FAILED`.
[config.json](https://github.com/user-attachments/files/16522665/config.json)

To create the Job with, I used the following command:
```
gcloud batch jobs submit example-job \
    --config=config.json \
    --location=us-central1
```

Reply:
I'm prepared a separate file with a description of the interesting points I've found out. 
I wrote two "issues" down in this file.
The document is [here](https://docs.google.com/document/d/1khwJ5sUzw2UDUARiF_h_qdvyXP5fmkC7yyOmvgCYXs0/edit?usp=sharing).

Reply:
Thanks, that's valuable feedback, I'll let the product team know about this.

As for the sample content - is there a reason you're using two runnables? I thought we'd go with a single runnable + a label dictionary, like in the previous 2 samples.

Reply:
There is no valid reason. Just to demonstrate how it can work with multiple runnables.
Let me change it to just one runnable.